components/blobserve/pkg/blobserve/blobserve.go

```diff
@@ -102,7 +106,19 @@ func (reg *Server) MustServe() {
 // serve serves a single file from an image
 func (reg *Server) serve(w http.ResponseWriter, req *http.Request) {
 	vars := mux.Vars(req)
-	repo, tag := vars["repo"], vars["tag"]
+	image := vars["image"]
+	pref, err := reference.ParseNormalizedNamed(image)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("cannot parse image '%s': %q", image, err), http.StatusNotFound)
+		return
+	}
+	repo := pref.Name()
+	var tag string
+	if refTagged, ok := pref.(reference.Tagged); ok {
+		tag = refTagged.Tag()
+	} else {
+		tag = "latest"
```

Comment:
much like before we should fail if there's no explicit tag. In combination with the caching that blobserve and registry-facade do, `latest` is too error prone.

Reply:
:+1:  _(The PathPrefix doesn't allow you to pass an image without a tag anyway. Thus, this part makes no sense here as well.)_